### ISSUE TITLE
Memory error when using `extract_points` method

### ISSUE THREAD

**Issue by Clemoreau** [pyvista/pyvista#1694]
Hello all, 

I've got an error when using `extract_points` method. This is a method I use very often without any problem. Basically, the code is:
```py
submesh = surf.extract_points(ind=group[1], adjacent_cells=False, include_cells=True)
```

What is different in comparison with usually is that `surf` (which is a pyvista PolyData) is containing a big amount of nodes (around 6 billions). I get this error, which looks like a memory error: (there are 5 885 588 points in surf)

Is it a limit of size linked with pyvista_ndarray ? Do you know this type of error ?

Thank you for you help

![image](https://user-images.githubusercontent.com/52700851/134904459-3e2903c0-1949-4b58-8144-8f305525966f.png)

I realize that the problem could be from `surf` variable (PolyData), rather than from `extract_points` method. When I display `surf` variable, I get this similar error:

![image](https://user-images.githubusercontent.com/52700851/134920434-f1613685-3887-4bad-9afd-ec8596ac221a.png)
 
-----

**System Information:**
--------------------------------------------------------------------------------
  Date: Mon Sep 27 13:49:27 2021 Paris, Madrid (heure d’été)

                OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 31.9 GiB
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Intel(R) HD Graphics 530
       GPU Version : 4.5.0 - Build 27.20.100.8681

  Python 3.8.8 (tags/v3.8.8:024d805, Feb 19 2021, 13:18:16) [MSC v.1928 64 bit
  (AMD64)]

           pyvista : 0.31.3
               vtk : 9.0.1
             numpy : 1.19.3
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.5
        matplotlib : 3.3.4
         pyvistaqt : 0.5.0
             PyQt5 : 5.15.2
           IPython : 7.22.0
             scipy : 1.6.2
              tqdm : 4.61.0
--------------------------------------------------------------------------------

**Comment by akaszynski** [comment]
```py
submesh = surf.extract_points(ind=group[1], adjacent_cells=False, include_cells=True)
```

Looks like the issue is with `ind=group[1]` where the indices of the source array aren't valid (for some reason).  Without more information, I can't tell, but 6 million points shouldn't result in a memory issue with 32GB RAM.

**Comment by Clemoreau** [comment]
Hello Alex, thank you for the quick response

However, I am pretty sure that `ind=group[1]` is not the problem for several reasons: 
- I checked manually that the indices were correct and still got the error
- The error doesn't always occur: with the same data in input, sometimes I get the error and sometimes, in debug mode, not.

That's why I thought it was an issue of memory. But I agree with you that with 32GB RAM, it shouldn't be a problem... 

Actually, I found out that `surf` which is a PolyData seems to be the problem because I am not able to visualize it (when I try, I got the same error, and my script crashes, meaning that I can't even debug it). So I assume that I'll have the same error everytime I will work with this PolyData variable, especially when using extract_points on it.

For your information, `surf` is extracted thanks to `extract_geometry` from a pyvista grid. This grid has 5885588 points and 16244 active scalars (these are the two numbers mentioned in the error). 
I can't send you the mesh corresponding to the grid because it is a confidential data, unfortunately. But let me know if I can give you more information to understand.

This is a script I used many times, without any error of this type, that's why I'm a bit confused about it. The only difference I see is the size of the mesh.

Thank you for your help !

**Comment by akaszynski** [comment]
> For your information, surf is extracted thanks to extract_geometry from a pyvista grid. This grid has 5885588 points and 16244 active scalars (these are the two numbers mentioned in the error).

Can you try using [extract_surface](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.extract_surface.html) instead?  There might be something weird going on with the extracted surface using the `extract_geometry` filter.

**Comment by Clemoreau** [comment]
Same error :(
![image](https://user-images.githubusercontent.com/52700851/135042526-130c117c-1390-4f70-b5ab-a07371088f9c.png)

**Comment by Clemoreau** [comment]
I come back to you to explain how I fixed my problem, if this can help anybody.

The error was due to the fact that my PolyData variable had a certain number of nodes (5885588 as mentioned in the error message) and a certain number of cells. And then I had a number of scalar data  (16244 as mentioned in the error message too) which didn't match with neither the number of nodes nor the number of cells. In fact I don't where these scalar values do come from, because they were imported via meshio.

I guess the error occurred when using extracts_points or any other extract method because it was impossible to know which scalar value to extract.

So I fixed the error by deleting the scalar values, that I didn't need anyway.

I hope my explanation is clear
